@here/harp-mapview/lib/geometry/TileGeometryCreator.ts

```diff
@@ -342,7 +342,17 @@ export class TileGeometryCreator {
         decodedTile: DecodedTile,
         textFilter?: (technique: Technique) => boolean
     ) {
+        if (
+            decodedTile.textPathGeometries === undefined &&
+            decodedTile.textGeometries === undefined
+        ) {
+            return;
+        }
+
         const mapView = tile.mapView;
+
+        mapView.createTextRenderer();
```

Comment:
Well, i think that's small violation of hollywood principle here.

TIle shouldn't know about these details of `MapView` implementation, i would rather add check in `MapView` that if there is some text to be rendererd to create `this.createTextRendererIfNeeded` ...

Not a major issue, but usuall smell coming from making some implementation detail public.

Reply:
Done, but the code looks like telescopic stick now.

Reply:
> telescopic stick

My general idea, was to collect really current status of text in renderLoop. It's not much, esp. Something like this in `MapView#render`

```

        let hasTextElements =
            this.m_overlayTextElements !== undefined && this.m_overlayTextElements.length > 0;
        renderList.forEach(({ zoomLevel, renderedTiles }) => {
            renderedTiles.forEach(tile => {
                // this exists already in code, we only need to add 
                this.renderTileObjects(tile, zoomLevel);
                // simple boolean sum over all tiles like this ...
                hasTextElements = hasTextElements || !tile.textElementGroups.empty();
                ...
            });
        });
        /// ...
        if (hasTextElements && !this.m_textElementsRenderer) {
            this.createTextElementsRendererIfNeeded();
        }
```

Reply:
That should be much more simple than my approach. Thanks.